### PR TITLE
Handle empty dataframes and single genomes in phylogeny functions

### PR DESCRIPTION
**Pull request by mmore500**
## Summary
Add robustness to phylogeny tree building and reconstruction functions by properly handling edge cases with empty datasets and single genomes. These changes prevent crashes and ensure valid output for boundary conditions.

## Key Changes
- **Empty dataframe handling**: Added early returns in `surface_postprocess_trie` and `_surface_postprocess_trie_via_pandas` to detect and handle empty input dataframes gracefully
- **Post-trunk-deletion empty check**: Added checks after trunk deletion to handle cases where the tree becomes empty after removing the trunk, properly cleaning up temporary columns
- **Regression tests**: Added comprehensive test coverage for edge cases:
  - `test_empty()` in both `test_surface_build_tree.py` and `test_surface_unpack_reconstruct.py` to verify empty genome sets produce empty phylogenies without crashing
  - `test_single_genome()` in both test files to verify single genome handling
  - `test_single_genome_no_delete_trunk()` in `test_surface_build_tree.py` to verify valid phylogeny output when trunk deletion is disabled

## Implementation Details
- Empty dataframe detection uses `df.lazy().limit(1).collect().is_empty()` for efficient checking without materializing the full dataframe
- Temporary columns (`is_trunk`, `ancestor_is_trunk`, `origin_time`) are cleaned up when dataframes become empty after trunk deletion
- Both Polars and Pandas code paths are updated consistently to handle these edge cases
- Logging added at each early return point for debugging and monitoring

https://claude.ai/code/session_01LsRsKbS6bk8QbqNpxYzyPc

### CODE REVIEW

**Review by mmore500 — COMMENTED** [review]
changes requested

File: hstrat/dataframe/_surface_postprocess_trie.py [hunk 1]
```diff
@@ -23,6 +23,22 @@
 )
 
 
+def _apply_empty_output_schema(df: pl.DataFrame) -> pl.DataFrame:
+    """Transform an empty trie DataFrame to match the postprocessed output
+    schema: add ``hstrat_rank`` and drop internal-only columns."""
+    assert df.is_empty(), "expected empty DataFrame"
+    df = df.with_columns(
+        hstrat_rank=pl.lit(None, dtype=pl.UInt64),
```
Comment (mmore500) [inline hstrat/dataframe/_surface_postprocess_trie.py]: just set ancestor_id= directly here too

**Review by mmore500 — COMMENTED** [review]
changes requested

**Review by mmore500 — COMMENTED** [review]
lgtm ready to merge after CI
